src/pages/top-up/Swap.tsx

```diff
@@ -135,16 +162,23 @@ export function Swap({ header }: Props): ReactElement {
     setSwapped(true)
 
     try {
-      await performSwap(desktopUrl, daiToSwap.toString)
+      await performSwapWithChecks(daiToSwap)
       const message = canUpgradeToLightNode
         ? 'Successfully swapped. Beginning light node upgrade...'
-        : 'Successfully swapped. Balances will refresh soon. You may now leave the page.'
+        : 'Successfully swapped. Balances will refresh soon. You may now navigate away.'
       enqueueSnackbar(message, { variant: 'success' })
 
       if (canUpgradeToLightNode) await restart()
     } catch (error) {
-      console.error(error) // eslint-disable-line
-      enqueueSnackbar(`Failed to swap: ${error}`, { variant: 'error' })
+      if (isSwapError(error)) {
+        enqueueSnackbar(error.snackbarMessage, { variant: 'error' })
+
+        if (error.originalError) {
+          console.error(error.originalError) // eslint-disable-line
+        }
+      } else {
+        console.error(error) // eslint-disable-line
```

Comment:
I would still show some snackbar even if it is not `SwapError`. It could be either a generic one something like "Ups, something went wrong", or you could add the `error.message`... 
Generally, it is a bad idea to have a case where you swallow an error without letting the user know (and IMHO `console.error` is not "letting the user know") especially if he expects some action...

Reply:
Yup, totally missed this case, thanks